x/txfees/keeper/txfee_filters/arb_tx.go

```diff
@@ -160,3 +185,68 @@ func isArbTxLooseSwapMsg(swapMsg poolmanagertypes.SwapMsgRoute, swapInDenom stri
 	swapInDenom = swapMsg.TokenInDenom()
 	return swapInDenom, false
 }
+
+func isWasmHooksExecutePayload(memo string) (map[string]interface{}, bool) {
+	jsonObject := make(map[string]interface{})
+	err := json.Unmarshal([]byte(memo), &jsonObject)
+	if err != nil {
+		return nil, false
+	}
+	wasm, ok := jsonObject["wasm"].(map[string]interface{})
+	if !ok {
+		return nil, false
+	}
+	_, ok = wasm["contract"].(string)
+	if !ok {
+		return nil, false
+	}
+	msg, ok := wasm["msg"].(map[string]interface{})
+	if !ok {
+		return nil, false
+	}
+	return msg, true
+}
+
+func getLoalIBCDenom(packet channeltypes.Packet, denom string) string {
+	if transfertypes.ReceiverChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel(), denom) {
+		// sender chain is not the source, unescrow tokens
+
+		// remove prefix added by sender chain
+		voucherPrefix := transfertypes.GetDenomPrefix(packet.GetSourcePort(), packet.GetSourceChannel())
+		unprefixedDenom := denom[len(voucherPrefix):]
+
+		// coin denomination used in sending from the escrow address
+		denom := unprefixedDenom
```

Comment:
Why shadowing the `denom` here? 
I would think we could only use denom and get rid off `unprefixedDenom`
```golang
denom = denom[len(voucherPrefix):] // or denom, _ = strings.CutPrefix(voucherPrefix)
denomTrace := transfertypes.ParseDenomTrace(denom)
if !denomTrace.IsNativeDenom() {
	denom = denomTrace.IBCDenom()
}
return denom
```

Reply:
again, mostly because it's what the internal ibc code does, so I want it to be as close to the source as possible

Reply:
Doesn't mean we can't do better ;) 
Jokes aside, I get your point... let's keep it as is